### PR TITLE
Adapt to LLVM 6 part 2

### PR DESCRIPTION
**Pull request by vchuravy**
Follow up with two missing changes to #26229

ccL @CodeLenz

### CODE REVIEW

**Review by vtjnash — APPROVED** [review]

File: src/jitlayers.cpp [hunk 1]
```diff
@@ -1048,13 +1048,19 @@ static void handleAllErrors(ArchiveWriterError E, HandlerT Handler) {
...
+    jl_safe_printf("ERROR: failed to emit output file %s\n", err.c_str());
+#else
 static void reportWriterError(ArchiveWriterError E) {
     std::string ContextStr = E.first.str();
     std::string err;
     if (!ContextStr.empty())
         err += ContextStr + ": ";
     err += E.second.message();
     jl_safe_printf("ERROR: failed to emit output file %s\n", err.c_str());
+#endif
```
Comment (ararslan) [inline src/jitlayers.cpp]: You could avoid having to write out the `jl_safe_printf` twice by moving this `#endif` above the call here

**Comment by CodeLenz** [comment]
I just checkout into vc/llvm6fixes and it compiled.

               _
   _       _ _(_)_     |  A fresh approach to technical computing
  (_)     | (_) (_)    |  Documentation: https://docs.julialang.org
   _ _   _| |_  __ _   |  Type "?help" for help.
  | | | | | | |/ _` |  |
  | | |_| | | | (_| |  |  Version 0.7.0-DEV.4522 (2018-03-09 18:50 UTC)
 _/ |\__'_|_|_|\__'_|  |  vc/llvm6fixes/1a7256830f* (fork: 2 commits, 0 days)
|__/                   |  x86_64-redhat-linux

julia> versioninfo()
Julia Version 0.7.0-DEV.4522
Commit 1a7256830f* (2018-03-09 18:50 UTC)
Platform Info:
  OS: Linux (x86_64-redhat-linux)
  CPU: Intel(R) Core(TM) i5-7300HQ CPU @ 2.50GHz
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-6.0.0 (ORCJIT, skylake)

Thank you !!

File: src/jitlayers.cpp (hunk 1 above)
Comment (vchuravy) [reply src/jitlayers.cpp]: I would rather move the `}` into the conditional branch since the function implementations are completely. different.

Comment (ararslan) [reply src/jitlayers.cpp]: ok

**Comment by CodeLenz** [comment]
@vchuravy, @vtjnash 

It seems that commit 

> 3086c23e605a84ac547fc3d5ab57729bba568295

Merge: 

> 3883debc79 793481037a

is breaking  the compilation of current master (Commit ead9eabe92* ) with both LLVM 5.0.1 and 6.0.0   (at least, this was the last commit that changed codegen.c, see error below). 

 ```
CC src/codegen.o
In file included from /home/lenz/julia/usr/include/llvm/ADT/EpochTracker.h:19:0,
                 from /home/lenz/julia/usr/include/llvm/ADT/SmallPtrSet.h:18,
                 from /home/lenz/julia/usr/include/llvm/Support/CommandLine.h:25,
                 from /home/lenz/julia/usr/include/llvm/CodeGen/MachinePassRegistry.h:23,
                 from /home/lenz/julia/usr/include/llvm/CodeGen/SchedulerRegistry.h:18,
                 from /home/lenz/julia/usr/include/llvm/CodeGen/TargetSubtargetInfo.h:22,
                 from /home/lenz/julia/src/codegen.cpp:42:
/home/lenz/julia/usr/include/llvm/Config/abi-breaking.h:23:6: warning: "LLVM_DISABLE_ABI_BREAKING_CHECKS_ENFORCING" is not defined, evaluates to 0 [-Wundef]
 #if !LLVM_DISABLE_ABI_BREAKING_CHECKS_ENFORCING
      ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /home/lenz/julia/src/intrinsics.cpp:7:0,
                 from /home/lenz/julia/src/codegen.cpp:1902:
/home/lenz/julia/src/ccall.cpp: In member function ‘std::__cxx11::string function_sig_t::generate_func_sig(const char*)’:
/home/lenz/julia/src/ccall.cpp:1328:50: error: no matching function for call to ‘copy(llvm::AttrBuilder&)’
                 paramattrs.push_back(std::copy(ab));
                                                  ^
In file included from /usr/include/c++/7/bits/char_traits.h:39:0,
                 from /usr/include/c++/7/string:40,
                 from /home/lenz/julia/src/codegen.cpp:29:
/usr/include/c++/7/bits/stl_algobase.h:446:5: note: candidate: template<class _II, class _OI> _OI std::copy(_II, _II, _OI)
     copy(_II __first, _II __last, _OI __result)
     ^~~~
/usr/include/c++/7/bits/stl_algobase.h:446:5: note:   template argument deduction/substitution failed:
In file included from /home/lenz/julia/src/intrinsics.cpp:7:0,
                 from /home/lenz/julia/src/codegen.cpp:1902:
/home/lenz/julia/src/ccall.cpp:1328:50: note:   candidate expects 3 arguments, 1 provided
                 paramattrs.push_back(std::copy(ab));
                                                  ^
In file included from /usr/include/c++/7/bits/locale_facets.h:48:0,
                 from /usr/include/c++/7/bits/basic_ios.h:37,
                 from /usr/include/c++/7/ios:44,
                 from /usr/include/c++/7/istream:38,
                 from /usr/include/c++/7/sstream:38,
                 from /home/lenz/julia/src/codegen.cpp:30:
/usr/include/c++/7/bits/streambuf_iterator.h:293:5: note: candidate: template<class _CharT> typename __gnu_cxx::__enable_if<std::__is_char<_Tp>::__value, std::ostreambuf_iterator<_CharT> >::__type std::copy(std::istreambuf_iterator<_CharT>, std::istreambuf_iterator<_CharT>, std::ostreambuf_iterator<_CharT>)
     copy(istreambuf_iterator<_CharT> __first,
     ^~~~
/usr/include/c++/7/bits/streambuf_iterator.h:293:5: note:   template argument deduction/substitution failed:
In file included from /home/lenz/julia/src/intrinsics.cpp:7:0,
                 from /home/lenz/julia/src/codegen.cpp:1902:
/home/lenz/julia/src/ccall.cpp:1328:50: note:   ‘llvm::AttrBuilder’ is not derived from ‘std::istreambuf_iterator<_CharT>’
                 paramattrs.push_back(std::copy(ab));
                                                  ^
In file included from /usr/include/c++/7/deque:66:0,
                 from /usr/include/c++/7/stack:60,
                 from /home/lenz/julia/usr/include/llvm/Support/FileSystem.h:42,
                 from /home/lenz/julia/usr/include/llvm/Object/SymbolicFile.h:22,
                 from /home/lenz/julia/usr/include/llvm/Object/ObjectFile.h:24,
                 from /home/lenz/julia/usr/include/llvm/Object/SymbolSize.h:14,
                 from /home/lenz/julia/src/codegen.cpp:50:
/usr/include/c++/7/bits/deque.tcc:985:5: note: candidate: template<class _Tp> std::_Deque_iterator<_Tp, _Tp&, _Tp*> std::copy(std::_Deque_iterator<_Tp, const _Tp&, const _Tp*>, std::_Deque_iterator<_Tp, const _Tp&, const _Tp*>, std::_Deque_iterator<_Tp, _Tp&, _Tp*>)
     copy(_Deque_iterator<_Tp, const _Tp&, const _Tp*> __first,
     ^~~~
/usr/include/c++/7/bits/deque.tcc:985:5: note:   template argument deduction/substitution failed:
In file included from /home/lenz/julia/src/intrinsics.cpp:7:0,
                 from /home/lenz/julia/src/codegen.cpp:1902:
/home/lenz/julia/src/ccall.cpp:1328:50: note:   ‘llvm::AttrBuilder’ is not derived from ‘std::_Deque_iterator<_Tp, const _Tp&, const _Tp*>’
                 paramattrs.push_back(std::copy(ab));
                                                  ^
In file included from /usr/include/c++/7/deque:64:0,
                 from /usr/include/c++/7/stack:60,
                 from /home/lenz/julia/usr/include/llvm/Support/FileSystem.h:42,
                 from /home/lenz/julia/usr/include/llvm/Object/SymbolicFile.h:22,
                 from /home/lenz/julia/usr/include/llvm/Object/ObjectFile.h:24,
                 from /home/lenz/julia/usr/include/llvm/Object/SymbolSize.h:14,
                 from /home/lenz/julia/src/codegen.cpp:50:
/usr/include/c++/7/bits/stl_deque.h:391:5: note: candidate: template<class _Tp> std::_Deque_iterator<_Tp, _Tp&, _Tp*> std::copy(std::_Deque_iterator<_Tp, _Tp&, _Tp*>, std::_Deque_iterator<_Tp, _Tp&, _Tp*>, std::_Deque_iterator<_Tp, _Tp&, _Tp*>)
     copy(_Deque_iterator<_Tp, _Tp&, _Tp*> __first,
     ^~~~
/usr/include/c++/7/bits/stl_deque.h:391:5: note:   template argument deduction/substitution failed:
In file included from /home/lenz/julia/src/intrinsics.cpp:7:0,
                 from /home/lenz/julia/src/codegen.cpp:1902:
/home/lenz/julia/src/ccall.cpp:1328:50: note:   ‘llvm::AttrBuilder’ is not derived from ‘std::_Deque_iterator<_Tp, _Tp&, _Tp*>’
                 paramattrs.push_back(std::copy(ab));
                                                  ^
make[1]: *** [Makefile:136: codegen.o] Error 1
make: *** [Makefile:76: julia-src-release] Error 2
```
After a make -C deps clean-llvm and commenting LLVM_VER in Make.user (to use LLVM 3.9.1) current master compiles without issues.

**Pull request merged**